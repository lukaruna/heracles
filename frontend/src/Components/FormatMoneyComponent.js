import React, {Component} from 'react'
import {formatMoney} from '../requesters/formatMoney'
class FormatMoneyComponent extends Component {

    state = {
        value: undefined,
        received: undefined
    }

    async handleSubmit(event) {
        event.preventDefault()
        try {
            const res = await formatMoney({value: this.state.value})
            this.setState({received: res.data.value})
        } catch {
            this.setState({received: "Error"})
        }
    }

    handleChange(event){
        this.setState({
            value: event.target.value
        })
    }

    render(){
        return (
            <form onSubmit = {this.handleSubmit.bind(this)}>
                <div>
                    <h1>Format money</h1>
                </div>
                <div>
                    <label htmlFor="numberInput">Input</label>
                    <input id="numberInput" type="number" onChange={this.handleChange.bind(this)}/>
                    <button type="submit">Send</button>
                </div>
                {
                    this.state.received === ("Error" || undefined)
                        ? <h2 test-id={"Error"} style={{'color': '#b22222'}}>{this.state.received}</h2>
                        : this.state.received && <h2 test-id={"ok"}>{this.state.received}</h2>
                }
            </form>
        )}
}

export default FormatMoneyComponent