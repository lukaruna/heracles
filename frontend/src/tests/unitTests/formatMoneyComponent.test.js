import React from 'react';
import FormatMoneyComponent from "../../Components/FormatMoneyComponent"
import {mount} from "../Enzyme"

let wrapper
beforeEach(()=>{
    wrapper = mount(<FormatMoneyComponent/>)
})

test('Match snapshot formatMoney component', () => {
    expect(wrapper).toMatchSnapshot()
});

test("Should correctly set the state value on change", () => {
    const value = 123;
    const instance = wrapper.instance()

    expect(wrapper.state('value')).toBe(undefined)
    jest.spyOn(instance, "handleChange")
    instance.handleChange({target: {value}})
    expect(instance.handleChange).toHaveBeenCalledWith({target: {value}})
    expect(wrapper.state('value')).toBe(value)
})

//TODO: tests for handleSubmit toHaveBeenCalled
//TODO: Snapshot test for error
