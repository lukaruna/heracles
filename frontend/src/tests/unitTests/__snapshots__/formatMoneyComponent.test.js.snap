// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot formatMoney component 1`] = `
<FormatMoneyComponent>
  <form
    onSubmit={[Function]}
  >
    <div>
      <h1>
        Format money
      </h1>
    </div>
    <div>
      <label
        htmlFor="numberInput"
      >
        Input
      </label>
      <input
        id="numberInput"
        onChange={[Function]}
        type="number"
      />
      <button
        type="submit"
      >
        Send
      </button>
    </div>
  </form>
</FormatMoneyComponent>
`;
